src/test/java/com/commercetools/pspadapter/notification/processor/impl/PaymentSalePendingProcessorTest.java

```diff
@@ -0,0 +1,84 @@
+import com.commercetools.pspadapter.facade.CtpFacade;
+import com.commercetools.service.ctp.CartService;
+import com.commercetools.service.ctp.OrderService;
+import com.commercetools.service.ctp.PaymentService;
+import com.commercetools.service.ctp.impl.PaymentServiceImpl;
+import com.google.gson.GsonBuilder;
+import com.paypal.api.payments.Event;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.payments.Payment;
+import io.sphere.sdk.payments.Transaction;
+import io.sphere.sdk.payments.TransactionState;
+import io.sphere.sdk.payments.TransactionType;
+import io.sphere.sdk.payments.commands.updateactions.ChangeTransactionState;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static com.commercetools.testUtil.CompletionStageUtil.executeBlocking;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyList;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PaymentSalePendingProcessorTest {
+
+    @Test
+    public void shouldCallUpdatePaymentWithCorrectArgs() {
+        // set up
+        Payment ctpMockPayment = mock(Payment.class);
+        Transaction transaction = mock(Transaction.class);
+        when(ctpMockPayment.getTransactions()).thenReturn(Collections.singletonList(transaction));
+        when(transaction.getType()).thenReturn(TransactionType.CHARGE);
+        when(transaction.getState()).thenReturn(TransactionState.SUCCESS);
+
+        NotificationProcessorBase processorBase = spy(new PaymentSalePendingProcessor(new GsonBuilder().create()));
+
+        doReturn(CompletableFuture.completedFuture(Optional.of(ctpMockPayment)))
+                .when(processorBase).getRelatedCtpPayment(any(), any());
+        SphereClient sphereClient = mock(SphereClient.class);
+
+        PaymentService paymentService = spy(new PaymentServiceImpl(sphereClient));
+
+        CtpFacade ctpFacade = spy(
+                new CtpFacade(mock(CartService.class), mock(OrderService.class), paymentService)
+        );
+
+        Event event = new Event();
+        event.setEventType(NotificationEventType.PAYMENT_SALE_PENDING.toString());
+
+        // test
+        doAnswer(invocation -> {
+            verifyUpdatePaymentCall(ctpMockPayment, invocation);
+            return CompletableFuture.completedFuture(ctpMockPayment);
+        }).when(paymentService).updatePayment(any(Payment.class), anyList());
+
+        Payment returnedPayment = executeBlocking(processorBase.processEventNotification(ctpFacade, event));
+
+        // assert
+        assertThat(returnedPayment).isEqualTo(ctpMockPayment);
```

Comment:
in fact it is the same instance. Doesn't have much sense. What exactly is expected to assert here?

Reply:
I assert that when I call the function `processEventNotification`, I got returned what I expected to be returned (e.g. the object is not modified or it's not a completely different object).

Reply:
so, i'm confused: 
  - you want to verify the returned object is _absolutely the same instance_? Then use `AbstractAssert#isSameAs()`
  - you want to verify the initial object is not modified? Then you are doing it wrong, because even if _ctpMockPayment_ is modified this assert will success.

Reply:
Honestly, the whole test method looks useless. For instance, the last assert: 
```java
verify(paymentService, times(1)).updatePayment(any(Payment.class), anyList());
```

You verify `updatePayment()` is called with any payment instance and any list? Does it have real sense, if this method can't be called in other way? Instead of that, guess, you need to verify the `updatePayment()` is called with some certain instance (maybe _ctpMockPayment_ ?) and the supplied list contains certain values (guess, `AddInterfaceInteraction` and `ChangeTransactionState` with `state == TransactionType.CHARGE`)

Reply:
I removed this part.

Reply:
Well, guess not the best choice to remove test completely. Maybe we just needed to finalize it correctly. As Uncle Bob says: it's better to have at least bad tests, rather than no tests at all.

Reply:
I put back one condition: https://github.com/commercetools/commercetools-paypal-plus-integration/pull/69/commits/9feaa1c95d0801909f6951a4035338107fd6153a

The other `verify` it's gone forever because as you wrote, it doesn't have much sense.

btw. don't forget what uncle Abraham said:
![Abraham](http://thefederalist.com/wp-content/uploads/2017/07/Lincoln.jpg)

Reply:
I don't believe everyone, i believe only Uncle Bob.

Or you think "no tests" is better than "some weird tests"?